crystals/lattice.py

```diff
@@ -107,6 +107,24 @@ def from_parameters(
         ValueError : if lattice parameters are invalid.
         """
         return cls(lattice_vectors_from_parameters(a, b, c, alpha, beta, gamma))
+    
+
+    @property
+    def direct_tensor(self):      
```

Comment:
Using `eval` is a big no-no. It is both slow, unsafe, and unnecessary in this case.

Looks like this `direct_tensor` can be derived from the row-wise lattice vectors, no?